simd/src/CMakeLists.txt

```diff
@@ -22,6 +22,16 @@ KOKKOS_ADD_LIBRARY(
   SOURCES ${SIMD_SOURCES}
   HEADERS ${SIMD_HEADERS}
 )
+
+# At least for SYCL+Cuda, we need to link with svml manually
```

Comment:
Should we restrict it to when NVIDIA GPU architectures are enabled then?

Reply:
What happens if we don't link against it?

Reply:
> What happens if we don't link against it?

See https://github.com/kokkos/kokkos/pull/6581#issuecomment-1796849653.

> Should we restrict it to when NVIDIA GPU architectures are enabled then?

My first instinct was that there is not much harm in linking against it if we detect support and it's unclear to me if there are other situations in which it's necessary.